src/control/character_controller.rs

```diff
@@ -546,28 +554,29 @@ impl KinematicCharacterController {
 
         let decomp = self.decompose_hit(translation_remaining, &hit.toi);
 
-        // An object is trying to slip if the tangential movement induced by its vertical movement
-        // points downward.
-        let slipping_intent = self.up.dot(&horiz_input_decomp.vertical_tangent) < 0.0;
+        // An object is trying to slip if its vertical (according to self.up) input motion points downward.
+        let slipping_intent = self.up.dot(&_vertical_input) < 0.0;
+
+        // An object is slipping if the tangential movement induced by its vertical movement points downward.
         let slipping = self.up.dot(&decomp.vertical_tangent) < 0.0;
 
-        // An object is trying to climb if its indirect vertical motion points upward.
-        let climbing_intent = self.up.dot(&input_decomp.vertical_tangent) > 0.0;
-        let climbing = self.up.dot(&decomp.vertical_tangent) > 0.0;
+        // An object is trying to climb if its vertical (according to self.up) input motion points upward.
+        let climbing_intent = self.up.dot(&_vertical_input) > 0.0;
+        // An object is climbing if the tangential movement induced by its vertical movement points upward.
+        let climbing = self.up.dot(&decomp.vertical_tangent) > 0.001;
 
-        let allowed_movement = if hit.is_wall && climbing && !climbing_intent {
+        let allowed_movement = if hit.is_wall && (climbing && !climbing_intent) {
             // Can’t climb the slope, remove the vertical tangent motion induced by the forward motion.
             decomp.horizontal_tangent + decomp.normal_part
```

Comment:
when blocking climb, we don't want to apply the nudge factor, right ?

Reply:
Then we still need to avoid climbing as a side effect 🤔 I think.

Not sure on the correct math to do that 😅 ; mayyybe we should cross that horizontal tangeant with self.up

Reply:
after testing, removing the nudge factor doesn't seem to fix the "can climb forbidden slopes", so I'm reverting that change. To be clear, the current behaviour is still better than what's on master.